### PR TITLE
Scott DeVoss

### PR DESCRIPTION
**Pull request by scottdevoss**
I thought it went well overall. I got to the first extension and did better than I thought I was going to do!

### CODE REVIEW

File: app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class ProjectsController < ApplicationController
+  def show
+    @project = Project.find(params[:id])
+
+    @years_of_experience = 0
```
Comment (juliet-e) [inline app/controllers/projects_controller.rb:5]: Really all you need in this controller action is line 3 - the rest of this logic should be encapsulated in a model method. 

You could create the following methods in your project model:
- contestant_count: The logic here would be `self.contestants.count`, it's a simple calculation but it's still a calculation so it should live at the model level per MVC conventions
- average_experience: You can use AR here! `self.contestants.average(:age)` and you can optionally add `.to_f.round(2)` to round it to 2 decimal spots

Remember that per MVC, we want slim controllers without much logic!

File: app/models/project.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 class Project < ApplicationRecord
   validates_presence_of :name, :material
   belongs_to :challenge
+  has_many :contestant_projects
+  has_many :contestants, through: :contestant_projects
```
Comment (juliet-e) [inline app/models/project.rb:5]: Your associations are set up great!

File: app/views/projects/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<h2>Project Name: <%= @project.name %></h2>
+<h2>Project Material: <%= @project.material %></h2>
+<h2>Theme: <%= @project.challenge.theme %></h2>
+<h2>Number of Contestants: <%= @project.contestants.count %></h2>
```
Comment (juliet-e) [inline app/views/projects/show.html.erb:4]: We don't want to use `.count` in the view either -- this should be calling whatever model methods you create to encapsulate the count and avg_experience methods -- so maybe @project.contestant_count and @project.avg_experience

File: spec/features/contestants/index_spec.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        expect(page).to have_content("Jay McCarroll")
+        expect(page).to have_content("Gretchen Jones")
+        expect(page).to have_content("Kentaro Kameyama")
+        expect(page).to have_content("Erin Robertson")
+      end
+      
+      it "Under each contestants name I see a list of projects (names) that they've been on" do
+        visit "/contestants"
+        
+        expect(page).to have_content("News Chic")
```
Comment (juliet-e) [inline spec/features/contestants/index_spec.rb:41]: This test doesn't quite verify that projects for listed for contestants just because we don't know where on the page these show up. To make sure we're see projects only for that contestant, we could:
1. use within blocks
2. use dynamic labels on our view to anchor the projects and then expect these labels, e.g. `expect(page).to have_content("Projects for Jay: News Chic")`

File: spec/features/projects/show_spec.rb [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+      #User Story 3
+      it "I see a count of the number of contestants on this project" do
+        visit "/projects/#{@news_chic.id}"
+
+        expect(page).to have_content("Number of Contestants: 2")
+
+        visit "/projects/#{@lit_fit.id}"
+
+        expect(page).to have_content("Number of Contestants: 0")
```
Comment (juliet-e) [inline spec/features/projects/show_spec.rb:47]: Nice testing here!

The edge case you covered for the case in which a project has no contestants could be solved if you encapsulate that logic at the model level and use AR. Then, you could also test this edge case at the model spec level and not worry about it at the feature test level (just test the happy path case with News Chic)
